optuna/samplers/_grid.py

```diff
@@ -261,7 +268,9 @@ def _same_search_space(self, search_space: Mapping[str, Sequence[GridValueType]]
                 return False
 
             for i, param_value in enumerate(search_space[param_name]):
-                if param_value != self._search_space[param_name][i]:
+                if not GridSampler._grid_value_equal(
```

Comment:
nit: Can we use `self` instead of `GridSampler` here? Other staticmethods in this class is called via `self` object (e.g. `_check_value`).

```suggestion
                if not self._grid_value_equal(
```

Reply:
Thanks, fixed.